Makefile

```diff
@@ -5,9 +5,19 @@ watch: ## Start a file watcher to run tests on change. (requires: watchexec)
 .PHONY: all
 all: lint test build ## test -> lint -> build
 
+.PHONY: deps
+deps: ## Setup or Update linters
+	go get -u github.com/kisielk/errcheck
+	go get -u golang.org/x/lint/golint
+	go get -u github.com/securego/gosec/cmd/gosec
+	go get -u honnef.co/go/tools/cmd/staticcheck
+	go mod tidy
+	go mod download
+	go mod verify
+
 .PHONY: test
 test: ## Runs the unit test suite
-	go test -race ./...
+	gotestsum --debug --format pkgname-and-test-fails -- -race ./...
```

Comment:
What do we need debug for? And the intention behind gotestsum was easy output, I think this should rather be something along `gotestsum -- -race`

Reply:
If there is a reason why you made it this way, please explain as an answer to this comment, and we can discuss the pros and cons.

Reply:
debug mode was enabled just to facilitate the users' debug log in testing, can be removed.
pkgname format was used because of the simplicity of the output and the package level distribution, and complete statistics in the end.
Default mode will print all the failures together in the end.

Reply:
Now it will print failures and errors when they occur AND in the end. Is that intended?

Reply:
Reference: https://github.com/gotestyourself/gotestsum#format
```
pkgname (default) - print a line for each package.
pkgname-and-test-fails - print a line for each package, and failed test output.
```
And hence, I choose `pkgname-and-test-fails`